cub/device/dispatch/dispatch_radix_sort.cuh

```diff
@@ -1113,6 +1363,141 @@ struct DispatchRadixSort :
 
     };
 
+    template <typename ActivePolicyT>
+    CUB_RUNTIME_FUNCTION __forceinline__
+    cudaError_t InvokeOnesweep()
+    {
+        typedef typename DispatchRadixSort::MaxPolicy MaxPolicyT;
+#if defined(__NVCOMPILER_CUDA__)
+        typedef OffsetT AtomicOffsetT;
+#else
+        typedef cuda::atomic<OffsetT, cuda::thread_scope_device> AtomicOffsetT;
+#endif
```

Comment:
There's still an issue using libcu++ atomics in this patch:

```
#if defined(__CUDA_ARCH__) && ((!defined(_MSC_VER) && __CUDA_ARCH__ < 600) || (defined(_MSC_VER) && __CUDA_ARCH__ < 700))
#  error "CUDA atomics are only supported for sm_60 and up on *nix and sm_70 and up on Windows."
#endif
```

Since this is host code, we need this to work across all SM versions.

It looks like we'll need to move the `AtomicOffsetT` into the per-SM policies, accounting for the different minimum version on windows. However, that check is at file-scope in `cuda\std\detail\__atomic`, so just including the atomic header while targeting older SMs is problematic.

I'll check with the libcu++ folks about this, maybe they know of a workaround.